### PR TITLE
Fix/et 833 opt in out js

### PR DESCRIPTION
**Pull request by juanfra**
🎫 https://moderntribe.atlassian.net/browse/ET-833
🎥 https://www.loom.com/share/7bd62a9b692e4cd3a581bf3ff1ae25c1

This one goes hand in hand with https://github.com/moderntribe/event-tickets/pull/1716 to complete the functionality. This is the implementation of the JS call, according to the data attributes defined on the other PR.

### CODE REVIEW

File: src/resources/js/v2/rsvp-block.js [hunk 1]
```diff
@@ -138,6 +135,35 @@ tribe.tickets.rsvp.block = {};
...
 	};
 
+	/**
+	 * Handle the RSVP toggle for listing in public attendee list.
+	 *
+	 * @since TBD
+	 *
+	 * @param {event} e submission event
+	 */
+	obj.handleDisplayToggle = function( e ) {
```
Comment (bordoni) [inline src/resources/js/v2/rsvp-block.js:145]: since we are using the variable, let us call it `event`.

File: src/resources/js/v2/rsvp-block.js [hunk 2]
```diff
@@ -138,6 +135,35 @@ tribe.tickets.rsvp.block = {};
...
+
+		const $input = $( e.target );
+		const rsvpId = $input.data( 'rsvp-id' );
+		const checked = $input.prop( 'checked' );
+		const attendeeIds = $input.data( 'attendee-ids' );
+		const nonce = $input.data( 'opt-in-nonce' );
+		const $container = e.data.container;
+
+		var data = {
+			action: 'tribe_tickets_rsvp_handle',
```
Comment (bordoni) [inline src/resources/js/v2/rsvp-block.js:156]: Wonder if we could store this action name in an attribute of this object, similar to how we approach our PHP.

File: src/resources/js/v2/rsvp-block.js [hunk 3]
```diff
@@ -138,6 +135,35 @@ tribe.tickets.rsvp.block = {};
...
+	 */
+	obj.handleDisplayToggle = function( e ) {
+		e.preventDefault();
+
+		const $input = $( e.target );
+		const rsvpId = $input.data( 'rsvp-id' );
+		const checked = $input.prop( 'checked' );
+		const attendeeIds = $input.data( 'attendee-ids' );
+		const nonce = $input.data( 'opt-in-nonce' );
+		const $container = e.data.container;
```
Comment (bordoni) [inline src/resources/js/v2/rsvp-block.js:153]: Should we save before using it or use it directly? (not sure)

File: src/resources/js/v2/rsvp-block.js [hunk 4]
```diff
@@ -138,6 +135,35 @@ tribe.tickets.rsvp.block = {};
...
+
+		var data = {
+			action: 'tribe_tickets_rsvp_handle',
+			ticket_id: rsvpId,
+			step: 'opt-in',
+			opt_in: checked,
+			opt_in_none: nonce,
```
Comment (paulmskim) [inline src/resources/js/v2/rsvp-block.js:160]: `opt_in_none`? Is this how it is handled in the PHP or is this a typo on the JS side?

Comment (juanfra) [reply src/resources/js/v2/rsvp-block.js:160]: No, that's a typo haha🤦‍♂️

File: src/resources/js/v2/rsvp-block.js (hunk 3 above)
Comment (juanfra) [reply src/resources/js/v2/rsvp-block.js:153]: I don't understand what you mean. What are you suggesting to save?

File: src/resources/js/v2/rsvp-block.js (hunk 2 above)
Comment (juanfra) [reply src/resources/js/v2/rsvp-block.js:156]: We already have it this way in all our different JS files across projects. I feel we should keep consistency with how we have the JS files and not PHP.

Comment (sc0ttkclark) [reply src/resources/js/v2/rsvp-block.js:156]: I'm also okay with leaving this as it is for now, but if there's anything we need to do to improve this later across our JS that'd be a cool ticket to work on

File: src/resources/js/v2/rsvp-block.js (hunk 3 above)
Comment (sc0ttkclark) [reply src/resources/js/v2/rsvp-block.js]: I'm good with leaving it set here

**Pull request merged**